internal/core/services/autoscaler/policies/roomoccupancy/policy.go

```diff
@@ -61,75 +61,98 @@ func NewPolicy(roomStorage ports.RoomStorage) *Policy {
 
 // CurrentStateBuilder fill the fields that should be considered during the autoscaling policy.
 func (p *Policy) CurrentStateBuilder(ctx context.Context, scheduler *entities.Scheduler) (policies.CurrentState, error) {
-	readyRoomsAmount, err := p.roomStorage.GetRoomCountByStatus(ctx, scheduler.Name, game_room.GameStatusReady)
+	readyCount, err := p.roomStorage.GetRoomCountByStatus(ctx, scheduler.Name, game_room.GameStatusReady)
 	if err != nil {
 		return nil, fmt.Errorf("error fetching ready game rooms amount: %w", err)
 	}
 
+	activeCount, err := p.roomStorage.GetRoomCountByStatus(ctx, scheduler.Name, game_room.GameStatusActive)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching active game rooms amount: %w", err)
+	}
+
+	occupiedCount, err := p.roomStorage.GetRoomCountByStatus(ctx, scheduler.Name, game_room.GameStatusOccupied)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching occupied game rooms amount: %w", err)
+	}
+
+	totalFreeSlots := (readyCount + activeCount + occupiedCount) * scheduler.MatchAllocation.MaxMatches
```

Comment:
Why are we counting occupied as a free slot? Shouldn't we count how many matches are actively running against the number of total possible matches?

Reply:
Oh, nvm, just saw the calculation below